sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -38,12 +38,14 @@ case class ShowTablesExec(
     val rows = new ArrayBuffer[InternalRow]()
 
     val tables = catalog.listTables(namespace.toArray)
+    // how to make sure the table is matched here? What is pattern and how to use it?
     tables.map { table =>
       if (pattern.map(StringUtils.filterPattern(Seq(table.name()), _).nonEmpty).getOrElse(true)) {
         rows += toCatalystRow(table.namespace().quoted, table.name(), isTempView(table))
       }
     }
 
+    // these rows are CatalystRow (which is InternalRow). How to convert it back to table in CatalogImpl?
```

Comment:
@cloud-fan

Reply:
This can only be done in `CatalogImpl`, we probably need to issue a `getTable` request for each table name to get detailed table information